pkg/mixin/search.go

```diff
@@ -0,0 +1,81 @@
+package mixin
+
+import (
+	"encoding/json"
+	"sort"
+	"strings"
+
+	"get.porter.sh/porter/pkg/printer"
+	"github.com/gobuffalo/packr/v2"
+	"github.com/pkg/errors"
+)
+
+// SearchOptions are options for searching remote mixins
+type SearchOptions struct {
+	Name string
+	printer.PrintOptions
+}
+
+// Searcher contains a packr.Box containing a searchable list of mixins
+type Searcher struct {
+	Box *packr.Box
+}
+
+// NewSearcher returns a Searcher with the provided packr.Box
+func NewSearcher(box *packr.Box) Searcher {
+	return Searcher{
+		Box: box,
+	}
+}
+
+// Validate validates the arguments provided to a search command
+func (o *SearchOptions) Validate(args []string) error {
+	err := o.validateMixinName(args)
+	if err != nil {
+		return err
+	}
+
+	return o.ParseFormat()
+}
+
+// validateMixinName validates either no mixin name is provided or only one is
+func (o *SearchOptions) validateMixinName(args []string) error {
+	switch len(args) {
```

Comment:
This []string{} name validation is common enough that I think we should make a resuable utility function out of it and use it everywhere. 😀  Not in your PR just saying I think I've seen it in at least 6 places.

Reply:
Yea, definitely.  I noticed that when writing this code.  One subtle difference is we allow `args` to have length of 0 (no name supplied).  Perhaps this means one more consolidated helper function... or an option on the one consolidated function.

Reply:
Search may be a one-off. Thanks for calling that out so that I didn't blindly think that it could use the same function as the others!